File: java/java_study/src/section01/Solution03.java
package section01;

public class Solution03 {

    /*
    * 문장 속 단어 (indexOf(), substring())
    *
    * 한 개의 문장이 주어지면 그 문장 속에서 가장 긴 단어를 출력하는 프로그램 작성
    *
    * input : it is time to study
    * output : study
    * */

    public static void main(String[] args) {

        System.out.println("result : " + solution("it is time to study"));

    }

    public static String solution(String str) {
        String answer = "";
        int max = Integer.MIN_VALUE;

        // split(" ")
//        String[] strArr = str.split(" ");
//        for (int i = 0; i < strArr.length; i++) {
//            int strLen = strArr[i].length();
//            if (strLen > max) {
//                max = strLen;
//                answer = strArr[i];
//            }
//        }

        // indexOf(), substring()
        int pos;
        // " "를 만나면 그 인덱스 번호를 반환, " "가 아니라면 -1 반환
        while ((pos = str.indexOf(" ")) != -1) {
            String tmp = str.substring(0, pos);
            int strLen = tmp.length();
            if (strLen > max) {
                max = strLen;
                answer = tmp;
            }
            str = str.substring(pos + 1);
        }

        // 마지막 단어 점검
        if (str.length() > max) {
            answer = str;
        }

        return answer;
    }
}

File: javascript/section03/solution03.js
/*
숫자만 추출

문자와 숫자가 섞여있는 문자열이 주어지면 그 중 숫자만 추출하여 그 순서대로 자연수를 만든다.
isNaN() : 숫자가 아니면 true (NaN : Not A Number)

input : g0en2T0s8eSoft (0208)
output : 208
*/

console.log(`result : ${solution("g0en2T0s8eSoft")}`);

function solution(str) {
    // 1) paresInt 사용
    // let answer = "";

    // for(let s of str) {
    //     if(!isNaN(s)) answer += s;
    // }

    // return parseInt(answer);

    // 2) 수학적 계산
    let answer = 0;

    for(let s of str) {
        if(!isNaN(s)) answer = answer * 10 + Number(s);
    }

    return answer;
}
